src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addLast(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Not in range");
+        }
+        if (index == size) {
+            addLast(value);
+        } else {
+            addBefore(value, index);
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> tempNode = getNode(index);
+        T result = tempNode.value;
+        tempNode.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> tempNode = getNode(index);
+        T result = tempNode.value;
+        if (size == 1) {
+            first = null;
+            last = null;
+            size--;
+            return result;
+        }
+        if (tempNode.previous == null) {
+            first = tempNode.next;
+            tempNode.next.previous = null;
+            size--;
+            return result;
+        }
+        if (tempNode.next == null) {
+            last = tempNode.previous;
+            last.next = null;
+            size--;
+            return result;
+        }
+        tempNode.previous.next = tempNode.next;
+        tempNode.next.previous = tempNode.previous;
+        size--;
+        return result;
     }
 
     @Override
     public boolean remove(T t) {
+        for (int i = 0; i < size; i++) {
+            T check = get(i);
```

Comment:
```suggestion
            T currentNode = get(i);
```

Reply:
Hmm.. get(i) returns not Node, but the value of that node, maybe better to use currentValue?

Reply:
yep, missed that, you can use currentValue